backend/fms_core/resources/sample.py

```diff
@@ -161,21 +161,29 @@ def import_obj(self, obj, data, dry_run):
 
         # TODO: This should throw a nicer warning if the individual already exists
         # TODO: Warn if the individual exists but pedigree/cohort is different
-        try:
-            individual, individual_created = Individual.objects.get_or_create(
-                name=get_normalized_str(data, "Individual ID"),
-                sex=get_normalized_str(data, "Sex", default=Individual.SEX_UNKNOWN),
-                taxon=taxon,
-                **({"pedigree": pedigree} if pedigree else {}),
-                **({"cohort": cohort} if cohort else {}),
-                **({"mother": mother} if mother else {}),
-                **({"father": father} if father else {}),
-            )
-            obj.individual = individual
-        except Exception as e:
-            individual_created = False
-            individual = None
-            errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
+
+        # If there's the individual's complete information then get/create
+        individual_is_complete = True if data["Individual ID"] and data["Sex"] and taxon else False
+        individual = None
+        if individual_is_complete:
+            try:
+                individual, individual_created = Individual.objects.get_or_create(
+                    name=get_normalized_str(data, "Individual ID"),
+                    sex=get_normalized_str(data, "Sex", default=Individual.SEX_UNKNOWN),
+                    taxon=taxon,
+                    **({"pedigree": pedigree} if pedigree else {}),
+                    **({"cohort": cohort} if cohort else {}),
+                    **({"mother": mother} if mother else {}),
+                    **({"father": father} if father else {}),
+                )
+                obj.individual = individual
+            except Exception as e:
+                individual_created = False
+                errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
+        elif data["Individual ID"] or data["Sex"] or taxon and not individual_is_complete:
+            errors["individual"] = ValidationError("In order to save the individual's information all the fields are required (Taxon, Sex and ID)", code="invalid")
```

Comment:
i would replace "... all the fields ..." by " ... these fields ...". In the context of the template, all the fields are not required. Also ID is not clear since on the template this info is in a "individual name" column.